src/background/actions/action-creator.ts

```diff
@@ -113,6 +116,8 @@ export class ActionCreator {
         this.registerTypeToPayloadCallback(Messages.Assessment.EnableVisualHelperWithoutScan, this.onEnableVisualHelperWithoutScan);
 
         this.registerTypeToPayloadCallback(Messages.Inspect.SetHoveredOverSelector, this.onSetHoveredOverSelector);
+
+        this.registerTypeToPayloadCallback(Messages.PathSnippet.AddPathForValidation, this.onAddPathForValidation);
```

Comment:
It seems that this new logic is not related to anything the else the `ActionCreator` currently holds. Do you mind creating a new action creator to separate the concerns here? In general, is good to follow Single Responsibility Principle, and one of the many ways to check for that is looking for subsets of functions that use disjunct subsets of class properties (like `pathSnippetActions` here). If you are able to find this subsets (of functions and properties) that points to the fact the class have more than one responsibility, as in this case.
I know that this class is already breaking the principle but it has been refactored on the pass to reduce this issue and it would be nice to not add to the tech debt here.

Reply:
Yes! That makes sense. I based the PathSnippet class off of the Inspect class because I figured they touched similar parts of the codebase, but it makes sense that we don't want to clutter the ActionCreator too much. I'll try abstracting out to its own ActionCreator. Thank you!